addons/mail/static/src/core/common/store_service.js

```diff
@@ -239,19 +241,16 @@ export const storeService = {
                                     if ([null, false, undefined].includes(val)) {
                                         return true;
                                     }
-                                    for (const v of collection) {
-                                        preinsert(v, r1, name, (r3) => {
-                                            l1.__list__.push(r3.localId);
-                                            r3.__invs__.add(r1.localId, name);
-                                        });
-                                    }
+                                    l1.push(...collection);
                                 } else {
```

Comment:
`push(...collection)` is equivalent, with added benefit that `push()` calls `onAdd()`